### PR TITLE
document that packaging needs to be installed before flash-attn

### PR DESCRIPTION
**Pull request by winglian**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -140,7 +142,8 @@ accelerate launch scripts/finetune.py examples/openllama-3b/lora.yml \
   git clone https://github.com/OpenAccess-AI-Collective/axolotl
   cd axolotl
 
-  pip3 install -e .
+  pip3 install packaging
+  pip3 install -e .[flash-attn]
   pip3 install protobuf==3.20.3
   pip3 install -U --ignore-installed requests Pillow psutil scipy
```
Comment (NanoCode012) [inline README.md:148]: You can merge these other installs together.

**Review by NanoCode012 — APPROVED** [review]

**Comment by NanoCode012** [comment]
Do we also need to mention that `ninja` is required?

File: README.md (hunk 1 above)
Comment (winglian) [reply README.md:148]: the problem is that if packaging isn't installed first, trying to install at the same time fails for flash-attn.

**Comment by winglian** [comment]
> Do we also need to mention that `ninja` is required?

I haven't seen that ninja has been problematic, but we can address in another pr

**Pull request merged**